### PR TITLE
chore(gamut-styles)!: deprecate legacy JS color variables 

### PR DESCRIPTION
**Pull request by jakemhiller**
# Overview

A more thorough overhaul of the colors to make way for a full migration to the new color set.

1. Moves the current colors (`standardColors`) to the `colors` export
2. Removes the extra keys from the `colors` export (a revert from the previous color PR)
3. Moves the old color swatches to `legacyColors`
4. Creates a new set of swatches to scale with the new colors, under the `swatches` export

This will allow us to use the `colors` export for Emotion components, instead of allowing all of the old colors as well.

Subsequent tasks:

1. Deprecate legacy SCSS variables (if needed) [GM-114]

### PR Checklist

- [ ] Related to Abstract designs:
- [ ] Related to JIRA ticket: [GM-113]
- [ ] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f453a4446b7ddd43dc0c220--gamut-preview.netlify.app>https://5f453a4446b7ddd43dc0c220--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f453a4446b7ddd43dc0c220>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@3.0.2-alpha.b8ea25.0<br/>@codecademy/gamut-styles@6.0.1-alpha.b8ea25.0<br/>@codecademy/gamut@16.0.2-alpha.b8ea25.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f492a6166cfbf8f868228a2--gamut-preview.netlify.app>https://5f492a6166cfbf8f868228a2--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f492a6166cfbf8f868228a2>Deploy Logs</a></p>

File: packages/gamut-styles/src/variables/deprecated-colors.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
+import { colors } from './colors';
+
+const { black, white } = colors;
+
+export const deprecatedColors = {
```
Comment (JoshuaKGoldberg) [inline packages/gamut-styles/src/variables/deprecated-colors.ts:5]: Missing a `@deprecated`?

Comment (jakemhiller) [reply packages/gamut-styles/src/variables/deprecated-colors.ts:5]: Yeah good call

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@3.0.2-alpha.953c19.0<br/>@codecademy/gamut-styles@6.0.1-alpha.953c19.0<br/>@codecademy/gamut@16.0.2-alpha.953c19.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f4ea048a33a7b00a950c6d1--gamut-preview.netlify.app>https://5f4ea048a33a7b00a950c6d1--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f4ea048a33a7b00a950c6d1>Deploy Logs</a></p>

**Pull request merged**